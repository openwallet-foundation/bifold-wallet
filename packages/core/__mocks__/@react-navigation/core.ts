const navigate = jest.fn()
const dispatch = jest.fn()
const replace = jest.fn()
const navigation = {
  __timestamp: process.hrtime(),
  navigate,
  replace,
  setOptions: jest.fn(),
  getParent: () => {
    return {
      navigate,
      dispatch,
      replace,
    }
  },
  getState: jest.fn(() => ({
    index: jest.fn(),
  })),
  goBack: jest.fn(),
  pop: jest.fn(),
  reset: jest.fn(),
  isFocused: () => true,
  dispatch,
}

const useNavigation = () => {
  return navigation
}

const useIsFocused = () => {
  return true
}

const CommonActions = {
  navigate: jest.fn(),
  reset: jest.fn(),
  goBack: jest.fn(),
}

export { useNavigation, useIsFocused, CommonActions }
