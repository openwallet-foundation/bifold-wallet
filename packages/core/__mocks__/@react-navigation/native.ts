const navigate = jest.fn()
const dispatch = jest.fn()
const replace = jest.fn()
const navigation = {
  __timestamp: process.hrtime(),
  navigate,
  replace,
  setOptions: jest.fn(),
  getParent: () => {
    return {
      navigate,
      dispatch,
      replace,
    }
  },
  getState: jest.fn(() => ({
    index: jest.fn(),
  })),
  goBack: jest.fn(),
  pop: jest.fn(),
  reset: jest.fn(),
  isFocused: () => true,
  dispatch,
}

const useNavigation = () => {
  return navigation
}

const useIsFocused = () => {
  return true
}

const useRoute = jest.fn().mockReturnValue({ params: {} })

const CommonActions = {
  navigate: jest.fn(),
  reset: jest.fn(),
  goBack: jest.fn(),
}

const useFocusEffect = jest.fn()
const createNavigatorFactory = jest.fn()

export { useNavigation, useIsFocused, useRoute, useFocusEffect, createNavigatorFactory, CommonActions }
