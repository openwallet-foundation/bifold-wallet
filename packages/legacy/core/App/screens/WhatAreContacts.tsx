import { NavigationProp, ParamListBase } from '@react-navigation/native'
import React from 'react'
import { useTranslation } from 'react-i18next'
import { SafeAreaView, ScrollView, StyleSheet, Text, View } from 'react-native'

import Link from '../components/texts/Link'
import { useTheme } from '../contexts/theme'
import { Screens, Stacks } from '../types/navigators'

interface WhatAreContactsProps {
  navigation: NavigationProp<ParamListBase>
}

const WhatAreContacts: React.FC<WhatAreContactsProps> = ({ navigation }) => {
  const { ColorPallet, TextTheme } = useTheme()
  const { t } = useTranslation()
  const styles = StyleSheet.create({
    container: {
      flex: 1,
      alignItems: 'center',
      backgroundColor: ColorPallet.brand.primaryBackground,
    },
    title: {
      ...TextTheme.headingTwo,
      marginBottom: 15,
    },
    pageContent: {
      marginTop: 30,
      paddingLeft: 25,
      paddingRight: 25,
    },
  })

  const goToContactList = () => {
    navigation.getParent()?.navigate(Stacks.ContactStack, { screen: Screens.Contacts })
  }

  const bulletPoints = [
    t('WhatAreContacts.ListItemDirectMessage'),
    t('WhatAreContacts.ListItemNewCredentials'),
    t('WhatAreContacts.ListItemNotifiedOfUpdates'),
    t('WhatAreContacts.ListItemRequest'),
  ].map((text, index) => {
    return (
      <View key={index} style={{ marginBottom: 10, flexDirection: 'row' }}>
        <Text style={{ ...TextTheme.normal, paddingRight: 5 }}>{'\u2022'}</Text>
        <Text style={[TextTheme.normal, { flexShrink: 1 }]}>{text}</Text>
      </View>
    )
  })

  return (
    <SafeAreaView style={styles.container}>
      <ScrollView
        contentContainerStyle={styles.pageContent}
        directionalLockEnabled
        automaticallyAdjustContentInsets={false}
        showsHorizontalScrollIndicator={false}
      >
        <Text style={styles.title} accessibilityRole="header">
          {t('WhatAreContacts.Title')}
        </Text>
        <Text style={TextTheme.normal}>{t('WhatAreContacts.Preamble')}</Text>
        {bulletPoints}
        <Text style={TextTheme.normal}>
          {`${t('WhatAreContacts.RemoveContacts')} `}
          <Link linkText={t('WhatAreContacts.ContactsLink')} onPress={goToContactList} />
        </Text>
      </ScrollView>
    </SafeAreaView>
  )
}

export default WhatAreContacts
