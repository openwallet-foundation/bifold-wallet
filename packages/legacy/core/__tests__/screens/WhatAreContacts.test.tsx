import { useNavigation } from '@react-navigation/native'
import { render } from '@testing-library/react-native'
import React from 'react'

import WhatAreContacts from '../../App/screens/WhatAreContacts'

describe('WhatAreContacts Screen', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  test('Renders correctly', async () => {
    const tree = render(<WhatAreContacts navigation={useNavigation()} />)
    expect(tree).toMatchSnapshot()
  })
})
