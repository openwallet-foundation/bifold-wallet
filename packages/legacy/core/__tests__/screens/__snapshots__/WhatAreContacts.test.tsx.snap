// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WhatAreContacts Screen Renders correctly 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#000000",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentContainerStyle={
      Object {
        "marginTop": 30,
        "paddingLeft": 25,
        "paddingRight": 25,
      }
    }
    directionalLockEnabled={true}
    showsHorizontalScrollIndicator={false}
  >
    <View>
      <Text
        accessibilityRole="header"
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 32,
            "fontWeight": "bold",
            "marginBottom": 15,
          }
        }
      >
        WhatAreContacts.Title
      </Text>
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 18,
            "fontWeight": "normal",
          }
        }
      >
        WhatAreContacts.Preamble
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginBottom": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 18,
              "fontWeight": "normal",
              "paddingRight": 5,
            }
          }
        >
          •
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 18,
                "fontWeight": "normal",
              },
              Object {
                "flexShrink": 1,
              },
            ]
          }
        >
          WhatAreContacts.ListItemDirectMessage
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginBottom": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 18,
              "fontWeight": "normal",
              "paddingRight": 5,
            }
          }
        >
          •
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 18,
                "fontWeight": "normal",
              },
              Object {
                "flexShrink": 1,
              },
            ]
          }
        >
          WhatAreContacts.ListItemNewCredentials
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginBottom": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 18,
              "fontWeight": "normal",
              "paddingRight": 5,
            }
          }
        >
          •
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 18,
                "fontWeight": "normal",
              },
              Object {
                "flexShrink": 1,
              },
            ]
          }
        >
          WhatAreContacts.ListItemNotifiedOfUpdates
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginBottom": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 18,
              "fontWeight": "normal",
              "paddingRight": 5,
            }
          }
        >
          •
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 18,
                "fontWeight": "normal",
              },
              Object {
                "flexShrink": 1,
              },
            ]
          }
        >
          WhatAreContacts.ListItemRequest
        </Text>
      </View>
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 18,
            "fontWeight": "normal",
          }
        }
      >
        WhatAreContacts.RemoveContacts 
        <Text
          accessibilityLabel="WhatAreContacts.ContactsLink"
          accessibilityRole="link"
          accessible={true}
          onPress={[Function]}
          style={
            Array [
              Object {
                "alignSelf": "flex-start",
                "color": "#FFFFFF",
                "fontSize": 18,
                "fontWeight": "normal",
                "textDecorationLine": "underline",
              },
              Object {},
            ]
          }
          testID="com.ariesbifold:id/WhatAreContacts.ContactsLink"
        >
          WhatAreContacts.ContactsLink
        </Text>
      </Text>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
